tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/PrettyWriter.scala

```diff
@@ -32,24 +33,42 @@ class PrettyWriter(writer: PrintWriter, textWidth: Int = 80, indent: Int = 2) ex
   // the stack of lambda declarations
   private var lambdaStack: List[TlaOperDecl] = Nil
 
-  def write(mod: TlaModule): Unit = {
-    writer.write(pretty(toDoc(mod), textWidth).layout)
-  }
+  def prettyWriteDoc(doc: Doc): Unit = writer.write(pretty(doc, textWidth).layout)
```

Comment:
The user of the `writeXXX` methods does not have to know about `Doc`, as you would see in the interface `TlaWriter`. Please revert this.

Reply:
Not sure I understand your concern here, before, all three `write` methods were implemented as `
writer.write(pretty( [X] , textWidth).layout)`, with varying `X`. All I did here was implement this as a separate method that takes `X` as the parameter.

Reply:
Check this new interface: https://github.com/informalsystems/apalache/blob/3dfa7a77e69762b0771a98d2ae591be9eab0be9d/tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/TlaWriter.scala

Reply:
Yes, all of these methods still exist and are unit

Reply:
Would you want `prettyWriteDoc` to be private?

Reply:
yes, or at least protected